Fixes bug 1289899 - Add query date range to signature report.

adngdb opened the pull request:
I side-tracked a bit while working on this, and removed an old, unused function as well as improved a bit the look of the signature report search form.

peterbe commented on webapp-django/crashstats/signature/jinja2/signature/signature_report.html:
```diff
@@ -13,21 +13,28 @@
+            Showing results from
+            <time class="ago" datetime="{{ query.start_date.isoformat() }}">{{ query.start_date | human_readable_iso_date }} UTC</time>
+            to <time class="ago" datetime="{{ query.end_date.isoformat() }}">{{ query.end_date | human_readable_iso_date }} UTC</time>.
+            <a href="#" class="toggle-filters">Edit filters</a>.
```
I don't like this. Sorry :)

It's confusing that it's labelled so differently from the "Show Search Filters". (ie. "edit" != "filter"). 
I think it's enough to have the "Show Search Filters" link in the upper right-hand corner.

peterbe commented on webapp-django/crashstats/signature/jinja2/signature/signature_report.html:
```diff
@@ -13,21 +13,28 @@
+        <p>
+            Showing results from
+            <time class="ago" datetime="{{ query.start_date.isoformat() }}">{{ query.start_date | human_readable_iso_date }} UTC</time>
+            to <time class="ago" datetime="{{ query.end_date.isoformat() }}">{{ query.end_date | human_readable_iso_date }} UTC</time>.
```
We talked about it somewhere else but let's get that fancy new jinja template function here.

peterbe commented on webapp-django/crashstats/signature/jinja2/signature/signature_report.html:
```diff
@@ -13,21 +13,28 @@
 {% block content %}
 <div id="mainbody">
     <div class="page-heading">
+        <a href="#" class="display-toggle-filters toggle-filters show">Show Search Filters</a>
```
This is an inline block preceding a block element. I think you should put this in a `<p>` tag and put `.display-toggle-filters` on it instead.

peterbe replied:
More importantly, can we make this a button instead?
It's a nit but it's odd that it's a hyperlink. Especially since you can't right-click and "Open in a New Window" on it. 

My webpages aren't perfect but I think the only time it's "OK" to have a hyperlink instead of a button is when it can actually be a link too. Like this:
```
<a href="{page_absolute_url}" onClick={showPermalinkWidget}>Permalink</a>
```

peterbe replied:
Also, I wonder about calling it "Search". **I** know it's a search because the underlying tech is SuperSearch. But the page is, after all, called "Signature report ..."
So perhaps "Show Filters" (and "Hide Filters").

peterbe commented on webapp-django/crashstats/signature/jinja2/signature/signature_report.html:
```diff
@@ -13,21 +13,28 @@
         >
             <button type="submit" id="search-button">Search</button>
-            <button class="new-line">new line</button>
+            <button class="new-line">New line</button>
```
Great!

peterbe commented on webapp-django/crashstats/signature/jinja2/signature/signature_report.html:
```diff
@@ -13,21 +13,28 @@
+            <time class="ago" datetime="{{ query.start_date.isoformat() }}">{{ query.start_date | human_readable_iso_date }} UTC</time>
+            to <time class="ago" datetime="{{ query.end_date.isoformat() }}">{{ query.end_date | human_readable_iso_date }} UTC</time>.
+            <a href="#" class="toggle-filters">Edit filters</a>.
+        </p>
```
By the way, this looks great!!

peterbe commented on webapp-django/crashstats/signature/views.py:
```diff
@@ -43,6 +43,43 @@
+    now = datetime.datetime.utcnow().replace(tzinfo=utc)
+
+    if not params.get('date'):
+        return (now - datetime.timedelta(days=7), now)
```
Perhas instead of `7` we can make it `settings.DEFAULT_SIGNATURE_REPORT_RANGE_DAYS`?

peterbe commented on webapp-django/crashstats/signature/views.py:
```diff
@@ -103,11 +140,12 @@ def signature_report(request, params, default_context=None):
+    context['query'] = {
+        'start_date': start_date,
+        'end_date': end_date,
+    }
```
I love this!

peterbe commented on webapp-django/crashstats/signature/static/signature/js/signature_report.js:
```diff
@@ -157,18 +157,16 @@ SignatureReport.init = function () {
 
         // Show or hide filters.
-        searchSection.on('click', '.toggle-filters', function (e) {
+        $('.toggle-filters').click(function (e) {
```
Ideally you should use the `.on('click'` instead. It's just a nit but I remember reading that all these hardcoded events lick `.click`, `.submit` are discouraged in favor of `.on('click', ` and `.on('submit', `

peterbe commented on webapp-django/crashstats/signature/static/signature/js/signature_report.js:
```diff
@@ -157,18 +157,16 @@ SignatureReport.init = function () {
+            // Update the main toggle link.
+            var mainToggleLink = $('.display-toggle-filters');
+            mainToggleLink.toggleClass('show');
+            mainToggleLink.html(
```
Should be `.text(`.

peterbe commented on webapp-django/crashstats/signature/static/signature/js/signature_report.js:
```diff
@@ -157,18 +157,16 @@ SignatureReport.init = function () {
+            var mainToggleLink = $('.display-toggle-filters');
+            mainToggleLink.toggleClass('show');
+            mainToggleLink.html(
+                (mainToggleLink.hasClass('show') ? 'Show' : 'Hide') + ' Search Filters'
```
English in the js code isn't great. It means we're now spelling "Search Filters" in two places. 
(Generally, it's bad because it's hard to I18N static .js files but easy to do in the template rendering. However, that's not a concern in our case). 
I think you should change the HTML to be 
```html
<a href="#" class="display-toggle-filters toggle-filters show" 
  data-opposite="Hide Search Filters">Show Search Filters</a>
```
And then use:
```javascript
var opposite = mainToggleLink.data.opposite;
mainToggleLink.data.opposite = mainToggleLink.text();
mainToggleLink.text(opposite);
```

peterbe commented on webapp-django/crashstats/signature/views.py:
```diff
@@ -43,6 +43,43 @@
 
 
+def get_date_range(params):
+    now = datetime.datetime.utcnow().replace(tzinfo=utc)
```
Replace with `now = timezone.now()` perhaps?

peterbe commented:
A couple of nits. I'm still a bit too tired to understand how this coincides with the (upcoming) changes in https://github.com/mozilla/socorro/pull/3421

adngdb commented:
@peterbe ^^^ that's a pretty big commit. I found out that I had already made a function to get date boundaries from a set of parameters, so I removed the recent one, moved the old one to a new file, and moved over my unit tests. That was a good idea: the tests found a bug in the old function, so I fixed that (nothing impacting though, it was just an edge case that was incorrectly handled). 

I have also changed the layout of the Signature report page so that tabs look like actual tabs, and not buttons. I find it much better now. 

And of course, I addressed all your comments! :)

adngdb commented on webapp-django/crashstats/topcrashers/tests/test_views.py:
```diff
@@ -41,15 +42,15 @@ def mocked_signature_first_date_get(**options):
                         "first_date": datetime.datetime(
                             2016, 5, 2, 0, 0, 0,
-                            tzinfo=UTC(),
+                            tzinfo=utc,
```
Mmm, sorry, I made these changes while I was working on the ``get_date_boundaries`` function, but then I ended up moving it somewhere else, so it's weird that this file has any changes. Luckily, it's all very simple.

peterbe commented:
There's a whitespace in front of the text "Show"
<img width="185" alt="screen shot 2016-08-11 at 12 45 52 pm" src="https://cloud.githubusercontent.com/assets/26739/17596928/96dea798-5fc1-11e6-976c-45c7d9f012ce.png">

peterbe commented on webapp-django/crashstats/signature/jinja2/signature/signature_report.html:
```diff
@@ -14,20 +14,26 @@
             <button type="submit" id="search-button">Search</button>
-            <button class="new-line">new line</button>
+            <button class="new-line">New line</button>
+            <h4>Search filters</h4>
```
"Search Filters" with a capital F.

peterbe commented:
r+ with nits.

peterbe commented on webapp-django/crashstats/crashstats/templatetags/jinja_helpers.py:
```diff
@@ -90,6 +90,17 @@ def timestamp_to_date(
+        )
+    )
+
+
```
If we only ever generate `<time class="ago">` from django, how come `timeutils.js` is iterating over `$('time.in')`. Where do they get created?

adngdb replied:
Ah, I haven't checked that. I assumed we had some somewhere deep in the site, but maybe it is not used anymore.

adngdb replied:
Well I just checked and we of course use in the Tokens page, to indicate when a token will expire! :)

peterbe replied:
Oh. I was too quick to merge. 

Sorry.
